homeassistant/auth/__init__.py

```diff
@@ -175,6 +220,45 @@ def auth_providers(self) -> List[AuthProvider]:
 
         await self._store.async_remove_credentials(credentials)
 
+    async def async_enable_user_mfa(self, user, mfa_module_id, data):
+        """Enable a multi-factor auth module for user."""
+        if mfa_module_id not in self._mfa_modules:
+            raise ValueError('Unable find multi-factor auth module: {}'
+                             .format(mfa_module_id))
+        if user.system_generated:
+            raise ValueError('System generated users cannot enable '
+                             'multi-factor auth module.')
+
+        module = self.get_auth_mfa_module(mfa_module_id)
+        if module.setup_schema is not None:
```

Comment:
Is this check necessary? The data is already generated based on an MFA module?

Reply:
`setup_schema` is optional for now. It is not validate the login form input, it is for the initial setup.